obspy/taup/tests/test_tau.py

```diff
@@ -107,6 +112,64 @@ def test_p_iasp91_manual(self):
         self.assertAlmostEqual(p_arrival.purist_distance, 35.00, 2)
         self.assertEqual(p_arrival.purist_name, "P")
 
+    @unittest.skipIf(not geobase.HAS_GEOGRAPHICLIB,
+                     'Module geographiclib is not installed')
+    def test_p_iasp91_geo_manual(self):
+        """
+        Manual test for P phase in IASP91 given geographical input.
+
+        This version of the test is used when geographiclib is installed
+        """
+        m = TauPyModel(model="iasp91")
+        arrivals = m.get_travel_times_geo(source_depth_in_km=10.0,
+                                          source_latitude_in_deg=20.0,
+                                          source_longitude_in_deg=33.0,
+                                          station_latitude_in_deg=55.0,
+                                          station_longitude_in_deg=33.0,
+                                          phase_list=["P"])
+        self.assertEqual(len(arrivals), 1)
+        p_arrival = arrivals[0]
+
+        self.assertEqual(p_arrival.name, "P")
+        self.assertAlmostEqual(p_arrival.time, 412.43, 2)
+        self.assertAlmostEqual(p_arrival.ray_param_sec_degree, 8.612, 3)
+        self.assertAlmostEqual(p_arrival.takeoff_angle, 26.74, 2)
+        self.assertAlmostEqual(p_arrival.incident_angle, 26.69, 2)
+        self.assertAlmostEqual(p_arrival.purist_distance, 35.00, 2)
+        self.assertEqual(p_arrival.purist_name, "P")
+
+    def test_p_iasp91_geo_fallback_manual(self):
+        """
+        Manual test for P phase in IASP91 given geographical input.
+
+        This version of the test checks that things still work when
+        geographiclib is not installed.
+        """
+        has_geographiclib_real = geobase.HAS_GEOGRAPHICLIB
+        geobase.HAS_GEOGRAPHICLIB = False
+        taup_geo.HAS_GEOGRAPHICLIB = False
+        tau.HAS_GEOGRAPHICLIB = False
```

Comment:
Yes, a comment is probably needed, and there may be a better way of doing it. 

What I am trying to do is test the code paths like: `taup.get_ray_paths_geo()` -> `taup. taup_geo.calc_dist()` -> `geodetics. gps2dist_azimuth()` -> `geodetics. calc_vincenty_inverse()`, which get called when `geographiclib` is not installed.  However, I want to run these tests even if `geographiclib` happens to be installed on the testing machine. One way of doing this is to run a test unconditionally that sets `HAS_GEOGRAPHICLIB` to `False` (whatever it started off as) and then return the variable to the initial state once the test is done. 

However, on a machine with `geographiclib` installed, I discovered that if I just set `obspy.geodetics.base.HAS_GEOGRAPHICLIB` to `False` the code in `obspy.taup.tau` behaves as if `HAS_GEOGRAPHICLIB` is `True`. If I set `obspy.taup.tau.HAS_GEOGRAPHICLIB` to `False`the code in `obspy.geodetics.base` behaves as if `HAS_GEOGRAPHICLIB` is `True`. Having written this down I find this behaviour so surprising I've gone back and checked. This really is what happens (at least on my machine running python 2.7).

I think this is down to a subtle difference between:

    from obspy.geodetics.base import HAS_GEOGRAPHICLIB
    HAS_GEOGRAPHICLIB = False

and:

    import obspy.geodetics.base
    obspy.geodetics.base.HAS_GEOGRAPHICLIB = False 

In the first case we add  `HAS_GEOGRAPHICLIB` to the local namespace (and the local dictionary that represents the local namespace), so updates do not propagate back. In the second we really use the namespace of the base module. Because we have imports like: `obspy.geodetics.base import HAS_GEOGRAPHICLIB` in `obspy.taup.taup_geo` and `obspy.tau.tau` we need to directly change the variable in those modules too. Changing the variable in one of these three places does not update the other two. I think this behaviour only happens when the module variable is immutable, and that you get a real reference if you do this with a list, for example. 

So, the only alternative I can think of would be to avoid `from obspy.geodetics.base import HAS_GEOGRAPHICLIB` everywhere. This will make the tests look nicer, but the rest of the code (where something like a read only variable is probably fine) more ugly.